File: JAVA/Media.java
package JAVA;

public class Media {
    private int id;
    private String judul;
    private int tahun;
    private String hakCipta;
    /* Constructor */
    public Media() {}

    public Media(int id, String judul, int tahun, String hakCipta) {
        this.id = id;
        this.judul = judul;
        this.setTahun(tahun);
        this.hakCipta = hakCipta;
    }
    
    /* Method */
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getJudul() { return judul; }
    public void setJudul(String judul) { this.judul = judul; }

    public int getTahun() { return tahun; }
    public void setTahun(int tahun) {    
        this.tahun = tahun;
    }

    public String getHakCipta() { return hakCipta; }
    public void setHakCipta(String hakCipta) { this.hakCipta = hakCipta; }
}

File: PHP/media.php
<?php

class Media {
    protected int $id;
    protected string $judul;
    protected int $tahun;
    protected string $hakCipta;

    public function __construct(int $id = 0, string $judul = "", int $tahun = 0, string $hakCipta = "") {
        $this->id = $id;
        $this->judul = $judul;
        $this->setTahun($tahun);
        $this->hakCipta = $hakCipta;
    }

    public function getId(): int { return $this->id; }
    public function setId(int $id): void { $this->id = $id; }

    public function getJudul(): string { return $this->judul; }
    public function setJudul(string $judul): void { $this->judul = $judul; }

    public function getTahun(): int { return $this->tahun; }
    public function setTahun(int $tahun): void { $this->tahun = max($tahun, 0); }

    public function getHakCipta(): string { return $this->hakCipta; }
    public function setHakCipta(string $hakCipta): void { $this->hakCipta = $hakCipta; }
}